apps/cowswap-frontend/src/modules/notifications/hooks/useTgSubscription.ts

```diff
@@ -1,35 +1,93 @@
-import { useCallback, useEffect, useMemo, useRef, useState } from 'react'
+import { useAtom } from 'jotai'
+import { createElement, ReactNode, useCallback, useEffect, useMemo, useRef, useState } from 'react'
 
 import { getCmsClient } from '@cowprotocol/core'
+import { useAddSnackbar } from '@cowprotocol/snackbars'
 
 import { TgAuthorization } from './useTgAuthorization'
 
+import { tgSubscriptionAtom } from '../atoms/tgSubscriptionAtom'
+
+const TG_DEV_BYPASS = process.env.NODE_ENV === 'development' && process.env.REACT_APP_TG_DEV_BYPASS === 'true'
+
+// In dev mode, store subscription state in sessionStorage to persist across calls
+const DEV_SUBSCRIPTION_KEY = 'tg_dev_subscription_state'
```

Comment:
Similar to atoms, I suggest to use the same pattern for storage keys: camelCase and versioned.